tests/testsuite/clean.rs

```diff
@@ -829,32 +847,39 @@ fn clean_dry_run() {
 
     // Start with no files.
     p.cargo("clean --dry-run")
-        .with_stdout("")
-        .with_stderr(
-            "[SUMMARY] 0 files\n\
-             [WARNING] no files deleted due to --dry-run",
-        )
+        .with_stdout_data("")
+        .with_stderr_data(str![[r#"
+[SUMMARY] [FILE_NUM] files
+[WARNING] no files deleted due to --dry-run
+
+"#]])
         .run();
     p.cargo("check").run();
     let before = p.build_dir().ls_r();
     p.cargo("clean --dry-run")
-        .with_stderr(
-            "[SUMMARY] [..] files, [..] total\n\
-             [WARNING] no files deleted due to --dry-run",
-        )
+        .with_stderr_data(str![[r#"
+[SUMMARY] [FILE_NUM] files, [FILE_SIZE]B total
+[WARNING] no files deleted due to --dry-run
+
+"#]])
         .run();
     // Verify it didn't delete anything.
     let after = p.build_dir().ls_r();
     assert_eq!(before, after);
-    let expected = itertools::join(before.iter().map(|p| p.to_str().unwrap()), "\n");
```

Comment:
https://github.com/rust-lang/cargo/actions/runs/9590054694/job/26444813493?pr=14096#step:11:4073

Paths on Windows didn't get normalized. It also happened here <https://github.com/rust-lang/cargo/pull/14091#discussion_r1644887976>.

Seems that expected string won't be normalized? https://docs.rs/snapbox/0.6.2/src/snapbox/assert/mod.rs.html#111-141

@epage, off the top of your head, what could be the cause?


```
--- Expected
++++ actual:   stdout
   1    1 | [ROOT]/foo/target
   2      - [ROOT]/foo/target\.rustc_info.json
   3      - [ROOT]/foo/target\CACHEDIR.TAG
   4      - [ROOT]/foo/target\debug
   5      - [ROOT]/foo/target\debug\.cargo-lock
   6      - [ROOT]/foo/target\debug\.fingerprint
   7      - [ROOT]/foo/target\debug\.fingerprint\bar-72b289f5cc241a07
   8      - [ROOT]/foo/target\debug\.fingerprint\bar-72b289f5cc241a07\dep-lib-bar
   9      - [ROOT]/foo/target\debug\.fingerprint\bar-72b289f5cc241a07\invoked.timestamp
  10      - [ROOT]/foo/target\debug\.fingerprint\bar-72b289f5cc241a07\lib-bar
  11      - [ROOT]/foo/target\debug\.fingerprint\bar-72b289f5cc241a07\lib-bar.json
  12      - [ROOT]/foo/target\debug\.fingerprint\foo-ad5acdf1e6368b18
  13      - [ROOT]/foo/target\debug\.fingerprint\foo-ad5acdf1e6368b18\dep-lib-foo
  14      - [ROOT]/foo/target\debug\.fingerprint\foo-ad5acdf1e6368b18\invoked.timestamp
  15      - [ROOT]/foo/target\debug\.fingerprint\foo-ad5acdf1e6368b18\lib-foo
  16      - [ROOT]/foo/target\debug\.fingerprint\foo-ad5acdf1e6368b18\lib-foo.json
  17      - [ROOT]/foo/target\debug\build
  18      - [ROOT]/foo/target\debug\deps
  19      - [ROOT]/foo/target\debug\deps\bar-72b289f5cc241a07.d
  20      - [ROOT]/foo/target\debug\deps\foo-ad5acdf1e6368b18.d
  21      - [ROOT]/foo/target\debug\deps\libbar-72b289f5cc241a07.rmeta
  22      - [ROOT]/foo/target\debug\deps\libfoo-ad5acdf1e6368b18.rmeta
  23      - [ROOT]/foo/target\debug\examples
  24      - [ROOT]/foo/target\debug\incremental
        2 + [ROOT]/foo/target/.rustc_info.json
        3 + [ROOT]/foo/target/CACHEDIR.TAG
        4 + [ROOT]/foo/target/debug/.cargo-lock
        5 + [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]/dep-lib-bar
        6 + [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]/invoked.timestamp
        7 + [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]/lib-bar
        8 + [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]/lib-bar.json
        9 + [ROOT]/foo/target/debug/.fingerprint/bar-[HASH]
       10 + [ROOT]/foo/target/debug/.fingerprint/foo-[HASH]/dep-lib-foo
       11 + [ROOT]/foo/target/debug/.fingerprint/foo-[HASH]/invoked.timestamp
       12 + [ROOT]/foo/target/debug/.fingerprint/foo-[HASH]/lib-foo
       13 + [ROOT]/foo/target/debug/.fingerprint/foo-[HASH]/lib-foo.json
       14 + [ROOT]/foo/target/debug/.fingerprint/foo-[HASH]
       15 + [ROOT]/foo/target/debug/.fingerprint
       16 + [ROOT]/foo/target/debug/build
       17 + [ROOT]/foo/target/debug/deps/bar-[HASH].d
       18 + [ROOT]/foo/target/debug/deps/foo-[HASH].d
       19 + [ROOT]/foo/target/debug/deps/libbar-[HASH].rmeta
       20 + [ROOT]/foo/target/debug/deps/libfoo-[HASH].rmeta
       21 + [ROOT]/foo/target/debug/deps
       22 + [ROOT]/foo/target/debug/examples
       23 + [ROOT]/foo/target/debug/incremental
       24 + [ROOT]/foo/target/debug
```

Reply:
It seems `cargo_test_support::paths::Path::ls_r()` gives non-normalized paths, but actual output.

Reply:
@dieterplex could you revert change in this specific test case, and put a `#[allow(deprecated)]`, so that we can unblock the entire PR?

Reply:
Done.